cpp/include/cudf/detail/get_value.cuh

```diff
@@ -48,11 +49,8 @@ T get_value(column_view const& col_view, size_type element_index, rmm::cuda_stre
   CUDF_EXPECTS(data_type(type_to_id<T>()) == col_view.type(), "get_value data type mismatch");
   CUDF_EXPECTS(element_index >= 0 && element_index < col_view.size(),
                "invalid element_index value");
-  T result;
-  CUDF_CUDA_TRY(cudaMemcpyAsync(
-    &result, col_view.data<T>() + element_index, sizeof(T), cudaMemcpyDefault, stream.value()));
-  stream.synchronize();
-  return result;
+  return cudf::detail::make_host_vector_sync(
+    device_span<T const>{col_view.data<T>() + element_index, 1}, stream).front();
```

Comment:
Quibble: In some alternate universe where pinned memory isn't being used and we drop back to the pageable resource, this is probably somewhat worse than what's there now.  I imagine that's not a case we really care about though.

Reply:
That's true. FWIW, we did a full benchmark run with a change like this in the `device_scalar`, and there was no negative impact from involving the heap unnecessarily.